identity-provider-service/src/bin/main.rs

```diff
@@ -1154,6 +1544,120 @@ fn extract_and_validate_request_query(
     })
 }
 
+/// Validate that the received version 1 request is well-formed.
+/// This check that all the cryptographic values are valid, and that the zero
+/// knowledge proofs in the request are valid.
+///
+/// The return value is either
+///
+/// - Ok(ValidatedRequest) if the request is valid or
+/// - Err(msg) where `msg` is a string describing the error.
```

Comment:
I don't think this response description is entirely accurate. The `err` case is more complex.